examples/jsm/transpiler/GLSLDecoder.js

```diff
@@ -780,6 +782,48 @@ class GLSLDecoder {
 
 	}
 
+	parseStructDefinition() {
+
+		const tokens = this.readTokensUntil( ';' );
+
+		const structName = tokens[ 1 ].str;
+		this.userDefinedStructTypes.add(structName);
+
+		if (tokens[ 2 ].str !== '{' ) {
+
+			throw new Error( 'Expected \'{\' after struct name ' );
+
+		}
+
+		const structMembers = [];
+		for ( let i = 3; i < tokens.length - 2; i += 3 ) {
+
+			const typeToken = tokens[ i ];
+			const nameToken = tokens[ i + 1 ];
+
+			if (typeToken.type != 'literal' || nameToken.type != 'literal') {
+				throw new Error( 'Invalid struct declaration' );
+			}
+
+			if (tokens[ i + 2 ].str !== ';' ) {
```

Comment:
You need to format the code. https://github.com/mrdoob/three.js/wiki/mr.doob's-code-style%E2%84%A2